### ISSUE TITLE
Segfault after handling large requests

### ISSUE THREAD

**Issue by Fryuni** [oven-sh/bun#5596]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.1.51 x86_64 unknown

### What steps can reproduce the bug?

When a large request is received a segfault happens when trying to reply to it.

Simple code:

```js
Bun.serve({
  async fetch(req) {
    const body = req.method === 'POST' ? await req.text() : null;

    console.log(`Body of length ${body?.length}`);

    const res = new Response(null, {
      status: 200,
      headers: {
        'Content-Type': 'application/json',
      }
    });

    console.log('Response built', res);

    return res;
  },
});
```

Send a request using HTTPie `http POST localhost:3000 @large-file.json` where `large-file.json` is [this file](https://github.com/json-iterator/test-data/blob/master/large-file.json), about 25MiB.

### What is the expected behavior?

The client should receive an empty 200 response and the server should continue running.

### What do you see instead?

```log
❯ bun run src/entryBun.ts
Body of length 26098894
Response built Response (0 KB) {
  ok: true,
  url: "",
  headers: Headers {},
  statusText: "",
  redirected: false,
  bodyUsed: false,
  status: 200
}
[1]    217776 segmentation fault (core dumped)  bun run src/entryBun.ts
```

The request is received correctly and the string on `body` is exactly the file, nothing is corrupted there.

The processing continues fine, the response is built correctly and the function returns. The client never receives the response and after a few seconds a segfault happens.

Happy to provide the core dump if no one can reproduce this.

### Additional information

If instead of `req.text()` I use `req.arrayBuffer()` than the first request works and the segfaul happens the second time I make the same request.

**Comment by Jarred-Sumner** [comment]
We fixed a crash that sounds similar to this in v1.0.2. Can you upgrade and see if it recurs?

**Comment by Fryuni** [comment]
Sure

**Comment by Fryuni** [comment]
Indeed, it was fixed on v1.0.2. I tried to find something similar on the issues but I found nothing.

Thanks :)
